DQM/TrackingMonitor/src/TrackAnalyzer.cc

```diff
@@ -2016,25 +2016,23 @@ void TrackAnalyzer::fillHistosForState(const edm::EventSetup& iSetup, const reco
       tkmes.TrackEtaPhiInverted->Fill(eta, -1 * phi);
       tkmes.TrackEtaPhiInvertedoutofphase->Fill(eta, 3.141592654 + -1 * phi);
       tkmes.TrackEtaPhiInvertedoutofphase->Fill(eta, -1 * phi - 3.141592654);
-      tkmes.TkEtaPhi_Ratio_byFoldingmap->getTH2F()->Divide(
-          tkmes.TrackEtaPhi->getTH2F(), tkmes.TrackEtaPhiInverted->getTH2F(), 1., 1., "");
-      tkmes.TkEtaPhi_Ratio_byFoldingmap_op->getTH2F()->Divide(
-          tkmes.TrackEtaPhi->getTH2F(), tkmes.TrackEtaPhiInvertedoutofphase->getTH2F(), 1., 1., "");
+      tkmes.TkEtaPhi_Ratio_byFoldingmap->divide(tkmes.TrackEtaPhi, tkmes.TrackEtaPhiInverted, 1., 1., "");
+      tkmes.TkEtaPhi_Ratio_byFoldingmap_op->divide(tkmes.TrackEtaPhi, tkmes.TrackEtaPhiInvertedoutofphase, 1., 1., "");
 
-      int nx = tkmes.TrackEtaPhi->getTH2F()->GetNbinsX();
-      int ny = tkmes.TrackEtaPhi->getTH2F()->GetNbinsY();
+      int nx = tkmes.TrackEtaPhi->getNbinsX();
+      int ny = tkmes.TrackEtaPhi->getNbinsY();
 
+      //NOTE: for full reproducibility when using threads, this loop needs to be
+      // a critical section
```

Comment:
How important is "full reproducibility when using threads"? (just thinking how soon a solution for that needs to found)

(e.g. are the divisions and get/setBinContent something that needs to be done on every event, or should they move to harvesting?)